### ISSUE TITLE
teslmate start fails due pqsql tables that not exists 

### ISSUE THREAD

**Issue by normanmaurer** [adriankumpf/teslamate#1707]
I followed the guide of a manual install on freebsd and all worked except when I try to start TeslaMate it fails with:

```
=====
===== LOGGING STARTED Wed Jun  9 15:57:40 CEST 2021
=====
ESC[22m2021-06-09 15:57:43.062 [info] System Info: Erlang/OTP 21
ESC[0mESC[22m2021-06-09 15:57:43.062 [info] Version: 1.23.3
ESC[0mESC[22m2021-06-09 15:57:43.111 [info] Application teslamate exited: TeslaMate.Application.start(:normal, []) returned an error: shutdown: failed to start child: TeslaMate.Api
    ** (EXIT) an exception was raised:
        ** (Postgrex.Error) ERROR 42P01 (undefined_table) relation "tokens" does not exist

    query: SELECT t0."id", t0."refresh", t0."access", t0."inserted_at", t0."updated_at" FROM "tokens" AS t0
            (ecto_sql 3.6.1) lib/ecto/adapters/sql.ex:749: Ecto.Adapters.SQL.raise_sql_call_error/1
            (ecto_sql 3.6.1) lib/ecto/adapters/sql.ex:682: Ecto.Adapters.SQL.execute/5
            (ecto 3.6.1) lib/ecto/repo/queryable.ex:224: Ecto.Repo.Queryable.execute/4
            (ecto 3.6.1) lib/ecto/repo/queryable.ex:19: Ecto.Repo.Queryable.all/3
            (teslamate 1.23.3) lib/teslamate/auth.ex:26: TeslaMate.Auth.get_tokens/0
            (teslamate 1.23.3) lib/teslamate/api.ex:101: TeslaMate.Api.init/1
            (stdlib 3.8.2.4) gen_server.erl:374: :gen_server.init_it/2
            (stdlib 3.8.2.4) gen_server.erl:342: :gen_server.init_it/6
ESC[0m{"Kernel pid terminated",application_controller,"{application_start_failure,teslamate,{{shutdown,{failed_to_start_child,'Elixir.TeslaMate.Api',{#{'__exception__' => true,'__struct__' => 'Elixir.Postgrex.Error',connection_id => 51915,message => nil,postgres => #{code => undefined_table,file => <<\"parse_relation.c\">>,line => <<\"1376\">>,message => <<\"relation \\"tokens\\" does not exist\">>,pg_code => <<\"42P01\">>,position => <<\"83\">>,routine => <<\"parserOpenTable\">>,severity => <<\"ERROR\">>,unknown => <<\"ERROR\">>},query => <<\"SELECT t0.\\"id\\", t0.\\"refresh\\", t0.\\"access\\", t0.\\"inserted_at\\", t0.\\"updated_at\\" FROM \\"tokens\\" AS t0\">>},[{'Elixir.Ecto.Adapters.SQL',raise_sql_call_error,1,[{file,\"lib/ecto/adapters/sql.ex\"},{line,749}]},{'Elixir.Ecto.Adapters.SQL',execute,5,[{file,\"lib/ecto/adapters/sql.ex\"},{line,682}]},{'Elixir.Ecto.Repo.Queryable',execute,4,[{file,\"lib/ecto/repo/queryable.ex\"},{line,224}]},{'Elixir.Ecto.Repo.Queryable',all,3,[{file,\"lib/ecto/repo/queryable.ex\"},{line,19}]},{'Elixir.TeslaMate.Auth',get_tokens,0,[{file,\"lib/teslamate/auth.ex\"},{line,26}]},{'Elixir.TeslaMate.Api',init,1,[{file,\"lib/teslamate/api.ex\"},{line,101}]},{gen_server,init_it,2,[{file,\"gen_server.erl\"},{line,374}]},{gen_server,init_it,6,[{file,\"gen_server.erl\"},{line,342}]}]}}},{'Elixir.TeslaMate.Application',start,[normal,[]]}}}"}
Kernel pid terminated (application_controller) ({application_start_failure,teslamate,{{shutdown,{failed_to_start_child,'Elixir.TeslaMate.Api',{#{'__exception__' => true,'__struct__' => 'Elixir.Postgre

```

Seems like the tables that it expect are not present. I didn't see any mention in the docs that I would need to create these myself. Any idea ?

**Comment by adriankumpf** [comment]
The database migrations need to be run before the application starts. In the Debian guide this is done via `ExecStartPre` in the systemd service definition:

```
ExecStartPre=/usr/src/teslamate/_build/prod/rel/teslamate/bin/teslamate eval "TeslaMate.Release.migrate"
```

Maybe @tuxbox can help us out. @tuxbox, is there something similar in FreebBSD service definitions?

**Comment by tuxbox** [comment]
this should not be a problem to fix this in the rc script at start up. I will make the change, test and then update the documentation. should be a quick fix.

**Comment by adriankumpf** [comment]
Merged  #1712 with the updated documentation.
